### PR TITLE
[DEV APPROVED] Upgrade to rails 4.2.10

### PR DESCRIPTION
**Pull request by scruti**
- Related to https://github.com/moneyadviceservice/comfortable-mexican-sofa/pull/30

This PR upgrades Rails from 4.1 to 4.2.

To do that, we followed the [Rails guide to upgrade from 4.1 to 4.2](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-4-1-to-rails-4-2) step by step and fixed issues that shown up after the upgrade.

I recommend to check the PR commit by commit to follow the specific changes that have been done during the upgrade process.

### CODE REVIEW

File: spec/support/shared_examples/controller_assigns_presented_site_files_shared_example.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    presented_files = assigns[:files].map(&:object)
+    sites_files = sites.map(&:files).flatten
+    expect(presented_files).to match_array(sites_files)
+  end
+
+  it 'responds success' do
+    get_index
+    expect(response).to be_success
+  end
+end
```
Comment (scruti) [inline spec/support/shared_examples/controller_assigns_presented_site_files_shared_example.rb]: wops, need to fix this.

File: spec/controllers/api/documents_controller_spec.rb [hunk 2]
```diff
@@ -142,10 +142,8 @@
         it 'returns pages with matching tags' do
           expect(documents.size).to be(2)
           expect(documents).to include(
-            hash_including(
-              'label' => 'Page with tag',
-              'label' => 'Page with another tag'
-            )
+            hash_including('label' => 'Page with tag'),
+            hash_including('label' => 'Page with another tag')
```
Comment (your) [inline spec/controllers/api/documents_controller_spec.rb]: 👍

File: spec/presenters/page_presenter_spec.rb [hunk 3]
```diff
@@ -49,7 +49,7 @@
       let(:language) { :en }
 
       it 'returns checked input' do
-        expect(language_input_tag).to include('input checked="checked"')
+        expect(language_input_tag).to include('input', 'checked="checked"')
```
Comment (your) [inline spec/presenters/page_presenter_spec.rb]: Awesome catch 👍

**Comment by scruti** [comment]
Update: I squashed multiple gem updates/changes into a single "update gem dependencies" first commit. The specific explanations for each case have been kept in the commit description.

**Pull request merged**